File: src/testFile1.ts
import {MemoryInfo, recordMemorySample} from "./types";

export function myFunction1(t: MemoryInfo): Uint8Array | null {
    if (t.foo === 0) {
        return null;
    }

    const len = t.foo * t.size;
    const mem = new Uint8Array(len);
    if (process.env.RECORD_MEMORY === "true") {
        recordMemorySample(mem);
    }

    return mem;
}

File: src/testFile1.c
#include "thing.h";

void* myFunction1(TestParameter* t) {
    if (t->foo == 0) {
        return 0;
    }

    size_t memory = t->foo * sizeof(size_t);
    void* mem = malloc(memory);
    recordMemorySample(mem, memory);

    return mem;
}


